lib/availability_resolver.js

```diff
@@ -273,14 +273,18 @@ class AvailabilityResolver {
       // first, check if there is a customer code and determine
       // requestability based on that.
       if (item.recapCustomerCode && item.recapCustomerCode[0]) {
-        const deliveryLocations = DeliveryLocationsResolver.__deliveryLocationsByCustomerCode(item.recapCustomerCode)
+        const deliveryLocations = DeliveryLocationsResolver.deliveryLocationsByRecapCustomerCode(item.recapCustomerCode)
         return deliveryLocations && deliveryLocations.length > 0
         // If there is no customer code, recap items default to physRequestable: true
       } else {
         return true
       }
     } else {
-      return requestableBasedOnHoldingLocation(item)
+      let deliveryLocations
+      if (item.m2CustomerCode && item.m2CustomerCode[0]) {
+        deliveryLocations = DeliveryLocationsResolver.deliveryLocationsByM2CustomerCode(item.m2CustomerCode)
+      }
+      return requestableBasedOnHoldingLocation(item) && deliveryLocations
```

Comment:
Is it possible for `deliveryLocations` to be an empty array? Should you also assert it's non-empty?

Reply:
[The most updated version of this branch](https://github.com/NYPL/discovery-api/blob/SCC-3415/m2-delivery-locations/lib/availability_resolver.js#L288) does make that check. not sure why you're seeing that there.